src/main/java/com/example/weather/client/exceptions/MyResourceNotFoundException.java

```diff
@@ -0,0 +1,12 @@
+package com.example.weather.client.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.NOT_FOUND)
```

Comment:
would be nice to provide some error message explaining why the request wasn't successful.